scripts/make_cross_border_flows.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-FileCopyrightText: Contributors to PyPSA-Eur <https://github.com/pypsa/pypsa-eur>
+#
+# SPDX-License-Identifier: MIT
+"""
+Export energy flows (imports and exports) at each snapshot.
+"""
+
+import logging
+
+import numpy as np
+import pandas as pd
+import pypsa
+
+from scripts._helpers import configure_logging, set_scenario_config
+
+logger = logging.getLogger(__name__)
+
+
+def extract_flows_timeseries(n: pypsa.Network) -> pd.DataFrame:
+    """
+    Extract all lines and links flows at each snapshot.
+
+    For lines (bidirectional):
+    - Positive flow: energy flows bus0 → bus1
+    - Negative flow: energy flows bus1 → bus0
+
+    For links (can be unidirectional or bidirectional):
+    - Unidirectional (p_min_pu >= 0): always flow bus0 → bus1
+    - Bidirectional (p_min_pu < 0):
+        - positive flows bus0 → bus1
+        - negative flows bus1 → bus0
+
+    Parameters
+    ----------
+    n : pypsa.Network
+        Solved PyPSA network.
+
+    Returns
+    -------
+    pd.DataFrame
+        Columns: snapshot, from_bus, to_bus, carrier, flow_MW
+    """
+    lines_flows = (
+        n.lines_t.p0.stack()
+        .reset_index()
+        .rename(columns={0: "flow_MW"})
+        .merge(
+            n.lines[["bus0", "bus1", "carrier"]],
+            left_on="Line",
+            right_index=True,
+        )
+        .assign(
+            positive_flow=lambda df: df["flow_MW"] >= 0,
+            from_bus=lambda df: np.where(df["positive_flow"], df["bus0"], df["bus1"]),
+            to_bus=lambda df: np.where(df["positive_flow"], df["bus1"], df["bus0"]),
+            flow_MW=lambda df: df["flow_MW"].abs(),
+        )[["snapshot", "from_bus", "to_bus", "carrier", "flow_MW"]]
+    )
+
+    links_flows = (
+        n.links_t.p0.stack()
```

Comment:
multiply `p0` with `snapshot_weightings` here. Otherwise only correct if model is at `1H` resolution